### PR TITLE
Feature/72 individual lookup

### PR DESCRIPTION
**Pull request by dheitzer**
[app#72](https://github.com/fecgov/fecfile-web-app/issues/72)

### CODE REVIEW

File: django-backend/fecfiler/contacts/views.py [hunk 1]
```diff
@@ -61,14 +62,39 @@ def committee_lookup(self, request):
...
+        max_fecfile_results_param = request.GET.get("max_fecfile_results", "")
+        if (
+            max_fecfile_results_param is not None
+            and max_fecfile_results_param.isnumeric()
+        ):
+            max_fecfile_results = int(max_fecfile_results_param)
+
+        fecfile_individuals = list(
+            self.get_queryset()
+            .filter(Q(type__icontains="IND") & (
```
Comment (mjtravers) [inline django-backend/fecfiler/contacts/views.py:91]: I believe this is an exact match with the field value so doesn't need a "contains". Would this work and be more exact?: .filter(Q(type="IND") & ...

File: django-backend/fecfiler/contacts/views.py [hunk 2]
```diff
@@ -61,14 +62,39 @@ def committee_lookup(self, request):
...
 
         return JsonResponse(return_value)
+
+    @action(detail=False)
+    def individual_lookup(self, request):
+        q = request.GET.get("q")
+        if q is None:
+            return HttpResponseBadRequest()
+
+        max_fecfile_results = 10
```
Comment (mjtravers) [inline django-backend/fecfiler/contacts/views.py:81]: Let's make this a constant outside the class and use that constant in both individual_lookup() and committee_lookup() functions

**Review by mjtravers — CHANGES_REQUESTED** [review]
A couple of comments put inline.

**Review by Elaine-Krauss-TCG — APPROVED** [review]

**Review by mjtravers — APPROVED** [review]

**Review by toddlees — APPROVED** [review]

**Comment by dheitzer** [comment]
Updates are ready for  CR.

File: django-backend/fecfiler/contacts/views.py [hunk 3]
```diff
@@ -76,14 +79,23 @@ def individual_lookup(self, request):
         if q is None:
             return HttpResponseBadRequest()
 
+        tokens = list(filter(None, re.split('[^\\w+]', q)))
+        if (not tokens) or (len(tokens) < 1):
+            return HttpResponseBadRequest()
+        term = ('.*' + '.* .*'.join(tokens) + '.*').lower()
```
Comment (toddlees) [inline django-backend/fecfiler/contacts/views.py:83]: I believe this brings us back into the world of the upper/lower problem.  I personally don't want to be in the business of designing search engines and I'd recommend that we stick to already developed solutions.

**Review by toddlees — CHANGES_REQUESTED** [review]

File: django-backend/fecfiler/contacts/views.py (hunk 3 above)
Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:83]: I believe you mentioned before @dheitzer that adding complexity to the querystring parsing proportionally adds unexpected behavior

**Review by Elaine-Krauss-TCG — APPROVED** [review]

File: django-backend/fecfiler/contacts/views.py (hunk 3 above)
Comment (dheitzer) [reply django-backend/fecfiler/contacts/views.py:83]: Yes that's a good point @toddlees.  I thought I would need to change to this based on discussion with @mjtravers and seeing how fec.gov seems to handle delimiters and Aurelia's feedback but I'm open to however we want to handle this.  Maybe we can discuss/review Tues when Matt's back?

Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:83]: image
image

Splitting up the search by tokens seems to make names with apostrophe's unmatchable.

Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:83]: It looks like fec splits up the tokens like you've got, so it's probably good that we match it in that sense.  If we can implement that without trading off matching names with `'`, that might be the way forward.

Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:83]: In a meeting with @MitchellTCG  we decided that we will pass on matching names with `'` until later

**Review by toddlees — APPROVED** [review]

**Pull request merged**